widgets/text/text_test.go

```diff
@@ -809,6 +809,66 @@ func TestTextDraws(t *testing.T) {
 				return ft
 			},
 		},
+		{
+			desc:   "tests maxContent length being applied - multiline",
+			canvas: image.Rect(0, 0, 10, 3),
+			opts: []Option{
+				MaxContent(10),
+				RollContent(),
+			},
+			writes: func(widget *Text) error {
+				return widget.Write("line0\nline1\nline2\nline3\nline4")
+			},
+			want: func(size image.Point) *faketerm.Terminal {
+				ft := faketerm.MustNew(size)
+				c := testcanvas.MustNew(ft.Area())
+				// Line return (\n) counts as one character
+				testdraw.MustText(c, "ine3", image.Point{0, 0})
+				testdraw.MustText(c, "line4", image.Point{0, 1})
+				testcanvas.MustApply(c, ft)
+				return ft
+			},
+		},
+		{
+			desc:   "tests maxContent length not being limited",
+			canvas: image.Rect(0, 0, 72, 1),
+			opts: []Option{
+				RollContent(),
+			},
+			writes: func(widget *Text) error {
+				return widget.Write("1234567890abcdefghijklmnopqrstuvwxyz1234567890abcdefghijklmnopqrstuvwxyz")
```

Comment:
Thank you for including test coverage.

The selected test data (content) is fairly long, which makes it harder to reason about what they are testing. Can we minimize its length to the base cases? (a content of 1, max content set to 1, etc...). Using shorter content and MaxContent should arguably be the same behavior as with longer values, but the tests should be easier to follow and verify.

Once we do that, let's make sure we cover all the cases:
- adding content and fitting within MaxContent exactly
- adding content and overwriting portion of the buffer
- adding content and overwriting the full buffer
- adding content that is itself longer than MaxContent

This is the full list that comes to mind. On top of this we should keep the test with newlines and also add at least one test that sets MaxContent and uses unicode characters that are wider than a single cell.

Please let me know if I can provide additional clarification.

Reply:
I'd simply gone for a longer string for testing just so it stood out and had some variance. As you've pointed out through, it's probably not helpful anyway.

I'll increase the test coverage and commit once ready, I agree that there's extra scenarios to test.

Thanks again for the feedback!

Reply:
Happy to provide the feedback, the thanks goes to you for the contribution!

Reply:
Unicode chars did force a bit of a rethink of the logic to perfect, but the code commit a few days ago should now be correct and the tests (hopefully) enough to cover all cases.

The CI test failure doesn't appear to be related to my changes, please let me know if there's anything else required here.